### PR TITLE
Admin page + Refresh button

### PR DESCRIPTION
**Pull request by wilmol**
- Changed stats endpoint to /stats/user
- Added endpoint for /stats/system (gets a few system stats + deadlocked phrases).
- Admin page to display system stats + deadlocked phrases.
- Also have the ability to resolve deadlocked phrases.
  - Added endpoint /phrase/{phraseId}/resolve
     - Votes on the phrase with userId 'RESOLVER'
  - Added endpoint /phrase/{phraseId}/votes/delete
     - Clears all votes for a phrase
   - 'RESOLVER' bypasses the check for completed phrases

Also added refresh button, works on any page (refreshes the API calls for that page).

### CODE REVIEW

File: concord-server/server/src/main/java/com/rabidgremlin/concord/resources/PhrasesResource.java [hunk 1]
```diff
@@ -183,16 +185,32 @@ public Response purgeCompletedPhrasesAndVotes(@ApiParam(hidden = true) @Auth Cal
...
+  @POST
+  @Timed
+  @Path("/{phraseId}/resolve")
+  public Response resolvePhrase(@ApiParam(hidden = true) @Auth Caller caller, @PathParam("phraseId") String phraseId, Label label)
+  {
+    String labelText = label.getLabel();
+    log.info("{} resolving {} as {}", caller, phraseId, labelText);
+
+    votesDao.upsert(phraseId, labelText, PHRASE_RESOLVER_USER_ID);
+    phrasesDao.markPhraseComplete(phraseId, labelText);
```
Comment (wilmol) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/resources/PhrasesResource.java]: This won't work, when we download the phrases (in `downloadCsv`) we only consider phrases which are not marked complete.

**Comment by wilmol** [comment]
@LaurenceTews @joedai @rabidgremlin @campbell18 
finished the deadlock phrases resolver page, also have phrase uploading feature ready.

**Comment by wilmol** [comment]
Tested resolving deadlocked phrases with real data

File: concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  private final Logger log = LoggerFactory.getLogger(GetEligiblePhrasesForCompletionFunction.class);
+
+  public GetEligiblePhrasesForCompletionFunction(List<GroupedPhraseVote> phraseVotes, Map<String, String> phrasesVotedOnByResolver, int consensusLevel)
+  {
+    this.phraseVotes = phraseVotes;
+    this.phrasesVotedOnByResolver = phrasesVotedOnByResolver;
+    this.consensusLevel = consensusLevel;
+  }
+
+  public List<Phrase> execute()
```
Comment (LaurenceTews) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java]: Uuge method. Could do with breaking it up with a few private methods.

File: concord-server/server/src/main/java/com/rabidgremlin/concord/dao/SystemStatsDao.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      "FROM votes " +
+      FILTER_USERS)
+  int getCountOfVotes();
+
+  @SqlQuery("SELECT COUNT(*) " +
+      "FROM labels")
+  int getCountOfLabels();
+
+  @SqlQuery("SELECT COUNT(*) " +
+      "FROM (SELECT userId FROM votes " + FILTER_USERS + " GROUP BY userId) ALIAS")
```
Comment (joedai) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/dao/SystemStatsDao.java]: Same here, a count distinct would do the trick

File: concord-ui/package.json [hunk 4]
```diff
@@ -8,13 +8,14 @@
...
-    "react-scripts": "2.1.1",
+    "react-scripts": "^2.1.8",
     "react-select": "^2.1.2",
     "redux": "4.0.1",
     "redux-thunk": "2.3.0",
     "rmwc": "4.0.6",
     "superagent": "4.0.0",
-    "webfontloader": "1.6.28"
+    "webfontloader": "1.6.28",
+    "@rmwc/list": "latest"
```
Comment (LaurenceTews) [inline concord-ui/package.json]: version should be pinned

Comment (wilmol) [reply concord-ui/package.json]: upgrading rmwc in the next PR, will fix it then

File: concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java (hunk 2 above)
Comment (wilmol) [reply concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java]: i didnt write this, tempted to rewrite

File: concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java [hunk 5]
```diff
@@ -74,10 +86,42 @@ public Response getUserStats(@ApiParam(hidden = true) @Auth Caller caller)
...
+
+    int totalPhrases = systemStatsDao.getTotalCountOfPhrases();
+    int completedPhrases = systemStatsDao.getCountOfCompletedPhrases();
+    int labelsUsed = systemStatsDao.getCountOfLabelsUsed();
+    int totalVotes = systemStatsDao.getCountOfVotes();
+    int totalLabels = systemStatsDao.getCountOfLabels();
+    int userCount = systemStatsDao.getCountOfUsers();
+
+    List<GroupedPhraseVoteWithMostRecentVoteTime> votedLabelsForUncompletedPhrases = votesDao.getLabelsForUncompletedPhrasesInVoteCountOrder();
+    Set<String> phraseIdsVotedOnByGodUser = votesDao.getVotesMadeByUser(RESOLVER_USER_ID).stream().map(PhraseLabel::getPhraseId).collect(Collectors.toSet());
```
Comment (joedai) [inline concord-server/server/src/main/java/com/rabidgremlin/concord/resources/StatsResource.java]: Minor: Probably good to rename it to phraseIdsVotedOnByResolver to make it consitent

File: concord-ui/src/components/systemstatstable.js [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    this.setState({ statsData: statsData });
+  }
+
+  clearVotes(phraseId) {
+    this.removeDeadLockedPhrase(phraseId);
+    this.props.dispatch(deleteVotesForPhrase(phraseId));
+  }
+}
+
+export default connect((state) => ({
```
Comment (LaurenceTews) [inline concord-ui/src/components/systemstatstable.js]: export default connect(({ systemStats, nextPhrase }) => ({
  error: systemStats.error,
  loading: systemStats.loading,
  statsData: systemStats.statsData,
  loadingResolvePhrase: nextPhrase.loading,
  errorResolvePhrase: nextPhrase.error
}))(SystemStatsTable);

would be a bit nicer.

File: concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java (hunk 2 above)
Comment (LaurenceTews) [reply concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java]: Your changes have made it too large to be readable though (~+20 lines), defs worth refactoring.

**Pull request merged**